metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -141,14 +141,12 @@ func (m *metricHints) CreateConfig(event bus.Event) []*common.Config {
 		moduleConfig["password"] = password
 	}
 
-	logp.Debug("hints.builder", "generated config: %v", moduleConfig.String())
```

Comment:
Hm, it's a pitty to loose this debug string. The object is created dynamically. If something goes wrong in `NewConfigFrom` one needs this debug output in order to tell what the input was.

MapStr has a few serialization methods: String, StringToPrint, and MarshalLogObject (used by zap loggers in case MapStr is passed directly). I wonder if we should: a) always apply the blacklist in these methods (there might be other places we write this debug string), or b) provide a special function/mapping serializing MapStr to a string.

We can also add a 'mixture' of a) and b) by implementing a `fmt.Formatter`. E.g. if `%v` or `%s` is given we apply the blacklist. But if the `+` or `#` flag is passed we do not apply the blacklist. For `MarshalLogObject` I would always prefer to apply the blacklist, as this is part of the logger.

@ycombinator @exekias WDYT?

Reply:
> We can also add a 'mixture' of a) and b) by implementing a `fmt.Formatter`. E.g. if `%v` or `%s` is given we apply the blacklist. But if the `+` or `#` flag is passed we do not apply the blacklist. For `MarshalLogObject` I would always prefer to apply the blacklist, as this is part of the logger.

I like this option the best.